packages/bits/src/lib/toolbar/toolbar-keyboard.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from "@angular/core";
+import { KEYBOARD_CODE } from "../../constants";
+import { MenuComponent } from "../menu";
+
+@Injectable()
+export class ToolbarKeyboardService {
+    public dynamicContainer: HTMLElement | null;
+
+    public moreBtn: HTMLButtonElement | null;
+
+    public menuComponent: MenuComponent | null = null;
+
+    public initService(container: HTMLElement, menu: MenuComponent | null): void {
+        this.dynamicContainer = container.querySelector(".nui-toolbar-content__dynamic");
+
+        if (this.dynamicContainer) {
+            this.moreBtn = this.dynamicContainer.querySelector(".nui-button.btn.menu-button");
+        }
+
+        if (menu) {
+            this.menuComponent = menu;
+        }
+    }
+
+    public disableFocusForMoreBtn(): void {
+        if (this.moreBtn) {
+            this.moreBtn.setAttribute("tabindex", "-1");
+        }
+    }
+
+    public onKeyDown(event: KeyboardEvent): void {
+        const { code } = event;
+
+        if (code === KEYBOARD_CODE.ARROW_LEFT || code === KEYBOARD_CODE.ARROW_RIGHT) {
+            event.preventDefault();
+            this.navigateByArrow(code);
+        }
+    }
+
+    private navigateByArrow(code: string): void {
```

Comment:
There's quite a bit of branching logic in this method, and it doesn't look like the unit tests we have cover most of the possibilities in here. It would be good to add tests to cover ~80% of the various conditions.

Reply:
the problem is that activeElement only readable property and I couldn't mock it. For testing this branching I need template, unfortunately, even toolbar template doesn't feet enough - debugElement doesn't have , moreButton, for instance. Honestly, I didn't find way how to test this lines of code correctly. By splitting test on two files  toolbar-keyboard.service.ts has approximately 80% line coverage. But it isn't good quality tests.

Reply:
Dima Firsov offer other solution, I applied it and ut.